core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/operators/aggregate/FinalAggregateOpExec.scala

```diff
@@ -57,7 +57,7 @@ class FinalAggregateOpExec[Partial <: AnyRef](
           }
           val fields: Array[Object] =
             (pair._1 ++ JavaConverters.asScalaBuffer(finalObject.getFields)).toArray
-          Tuple.newBuilder().add(schema, fields).build()
+          Tuple.newBuilder(schema).addSequentially(fields).build()
```

Comment:
Can you please add a comment on the addSequentially function in Tuple.java file explaining its usage.

Reply:
Also I saw that we are calculating the schema in lines 51-57. Ideally we shouldn't because schema is already being provided by OpDesc. I realize that pushing schema into OpExec may be difficult for Aggregation. If that is the case, please add a TODO line above that someone else will take care of later. The TODO can just say that we have to use the already built schema.

Reply:
Looking at the code above, we are indeed only calculating the schema object once. I think it's a bit hard to use the opDesc schema here since this is a generic aggregateOpExec. this implementation is good for now.

Reply:
Added comments. I did find it challenging to work with tuples that were temporary or rather the ones that weren't actually input or output tuples. But I think that was the case only in aggregate and progressive.